pkgs/stdenv/generic/default.nix

```diff
@@ -77,6 +77,8 @@ let
 
   stdenv = (stdenv-overridable argsStdenv);
 
+  buildCanExecuteHost = buildPlatform.canExecute hostPlatform;
```

Comment:
I view this file as almost completely oblivious of internals of `buildPlatform` or `hostPlatform`. There is one exception (which we would probably want to remove eventually in favour of explicit `stdenv.hostPlatform.isx86_64` & co.):

```nix
      inherit (hostPlatform)
        isDarwin isLinux isSunOS isCygwin isBSD isFreeBSD isOpenBSD
        isi686 isx86_32 isx86_64
        is32bit is64bit
        isAarch32 isAarch64 isMips isBigEndian;
```

There is certain charm of being able to grep for `hostPlatform` or `buildPlatform` to see where we peek at the implementation details when building things in individual packages.

The above 2 points very weakly push me to say that I prefer explicit `stdenv.buildPlatform.canExecute stdenv.hostPlatform` slightly more than external helper.

But then again I almost never use it myself.

Reply:
> There is certain charm of being able to grep for `hostPlatform` or `buildPlatform`

This is an excellent point.  I have pushed another commit which changes the name to `buildPlatformCanExecuteHostPlatform`.  It's more verbose, but greppability matters.

And yes, I think those `inherit (hostPlatform)`s should be deleted too.